doc/notes/cost-model-representations/cost-model-representations.md

```diff
@@ -0,0 +1,216 @@
+# Representations of the Plutus Core cost model
```

Comment:
Can we move any of this to Haddock on the appropriate bits of the codebase?

Reply:
> Can we move any of this to Haddock on the appropriate bits of the codebase?

I've done that for some of the stuff here, but I gave up after some time because this document is mostly a summary of stuff that's spread out over numerous Haddock comments in different parts of the codebase and I was just finding myself repeating stuff that's already there.  I think the Haddock's a bit problematic because it's hard to know where to start: the point of this document is to give you an overview of the whole thing, and splitting it up and putting it back in the Haddock kind of defeats the purpose.  Maybe we need a big giant Haddock comment in a single place, with links to relevant bits of code. I'm not sure where would be a sensible place to put that though.